File: iOS/Code/Xamarin/bugTrap/bugTrapKit/bugTrapKit.cs
using UIKit;
using ServiceStack;

namespace bugTrapKit
{
	public sealed class bugTrapKit
	{
		static readonly bugTrapKit shared = new bugTrapKit { BtWindow = new BtOverlayWindow () };

		public static bugTrapKit Shared {
			get {
				return shared;
			}
		}

		internal BtOverlayWindow BtWindow;

		bugTrapKit ()
		{

		}

		public void Init ()
		{
//			TrapState.Shared.InActionExtension = true;

			shared.BtWindow.Frame = UIScreen.MainScreen.Bounds;
			shared.BtWindow.WindowLevel = UIWindowLevel.StatusBar;
			shared.BtWindow.MakeKeyAndVisible();

			EnsureAppearance();
		}

		void EnsureAppearance ()
		{

		}

		public void ToggleTriggerVisibility ()
		{
			var root = shared.BtWindow.RootViewController as BtOverlayViewController;

			if (root != null) root.ToggleTriggerVisibility();
		}


		public static UIViewController GetInitialController ()
		{
			// Configure ServiceStack Exprot Client
			IosPclExportClient.Configure();

			TrackerService.Shared.Authenticate();

			return UIStoryboard.FromName("bugTrapKit", null).InstantiateInitialViewController();
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/bugTrapKit.swift
//
//  bugTrapKit.swift
//  bugTrap
//
//  Created by Colby Williams on 8/21/15.
//  Copyright © 2015 bugTrap. All rights reserved.
//

import Foundation

public class bugTrapKit {
	
	public class var Shared : bugTrapKit {
		
		struct Static {
			static let instance: bugTrapKit = bugTrapKit(window: BtOverlayWindow())
		}
		
		return Static.instance
	}
	
	internal let btWindow: BtOverlayWindow
	
	internal init (window: BtOverlayWindow) {
		btWindow = window
	}
	
	public func Start() {
		btWindow.frame = UIScreen.mainScreen().bounds
		btWindow.windowLevel = UIWindowLevelStatusBar
		btWindow.makeKeyAndVisible()
	}
	
	public func ToggleTriggerVisibility() {
		if let root = btWindow.rootViewController as? BtOverlayViewController {
			root.toggleTriggerVisibility()
		}
	}
}